### ISSUE TITLE
torch.compile accuracy issue when graph outputs are DTensors with _Partial placements

### ISSUE THREAD

**Issue by bdhirsh** [pytorch/pytorch#119485]
This is a placeholder for an internal issue, no repro yet.

(1) When torch.compile'ing a model, some of inputs to some of the graphs are DTensors with the Partial placement, and the net result is bad accuracy during training

(2) When manually coercing these DTensors to the Replicate() placement when they are the output of a previous graph, this appears to fix accuracy issues.

I'm hoping I can try to create a self-contained repro of this at some point.

cc @ezyang @gchanan @zou3519 @kadeng @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @msaroufim @albanD @anijain2305

**Comment by soulitzer** [comment]
Marking as high-pri because this is an accuracy/correctness issue

**Comment by bdhirsh** [comment]
This was fixed by https://github.com/pytorch/pytorch/pull/118670
